systemtest/src/test/java/io/strimzi/systemtest/migration/MigrationST.java

```diff
@@ -0,0 +1,265 @@
+        );
+
+        ClientUtils.waitForClientsSuccess(testStorage.getProducerName(), testStorage.getConsumerName(), testStorage.getNamespaceName(), testStorage.getMessageCount());
+
+        LOGGER.info("5. finishing migration - applying {}: enabled annotation, controllers should be rolled", Annotations.ANNO_STRIMZI_IO_KRAFT);
+        KafkaResource.replaceKafkaResourceInSpecificNamespace(testStorage.getClusterName(),
+            kafka -> kafka.getMetadata().getAnnotations().put(Annotations.ANNO_STRIMZI_IO_KRAFT, "enabled"), testStorage.getNamespaceName());
+
+        controllerPodsSnapshot = RollingUpdateUtils.waitTillComponentHasRolledAndPodsReady(testStorage.getNamespaceName(), controllerSelector, 3, controllerPodsSnapshot);
+
+        KafkaUtils.waitUntilKafkaStatusContainsKafkaMetadataState(testStorage.getNamespaceName(), testStorage.getClusterName(), KafkaMetadataState.KRaft.name());
+
+        LOGGER.info("6. removing LMFV and IBPV from Kafka config -> brokers and controllers should be rolled");
+        KafkaResource.replaceKafkaResourceInSpecificNamespace(testStorage.getClusterName(), kafka -> {
+            kafka.getSpec().getKafka().getConfig().remove("log.message.format.version");
+            kafka.getSpec().getKafka().getConfig().remove("inter.broker.protocol.version");
+        }, testStorage.getNamespaceName());
+
+        RollingUpdateUtils.waitTillComponentHasRolledAndPodsReady(testStorage.getNamespaceName(), brokerSelector, 3, brokerPodsSnapshot);
+        RollingUpdateUtils.waitTillComponentHasRolledAndPodsReady(testStorage.getNamespaceName(), controllerSelector, 3, controllerPodsSnapshot);
+
+        LOGGER.info("7. creating KafkaTopic: {} and checking if the metadata are only in KRaft", kraftTopicName);
+        resourceManager.createResourceWithWait(extensionContext, KafkaTopicTemplates.topic(testStorage.getClusterName(), kraftTopicName, testStorage.getNamespaceName()).build());
+
+        LOGGER.info("7a. checking if metadata about KafkaTopic: {} are in KRaft controller", kraftTopicName);
+
+        assertThatTopicIsPresentInKRaftMetadata(controllerSelector, kraftTopicName);
+
+        LOGGER.info("7b. checking if we are able to do a message transmission on KafkaTopic: {}", kraftTopicName);
+
+        immediateClients = new KafkaClientsBuilder(immediateClients)
+            .withTopicName(kraftTopicName)
+            .build();
+
+        resourceManager.createResourceWithWait(extensionContext,
+            immediateClients.producerTlsStrimzi(testStorage.getClusterName()),
+            immediateClients.consumerTlsStrimzi(testStorage.getClusterName())
+        );
+
+        ClientUtils.waitForClientsSuccess(testStorage.getProducerName(), testStorage.getConsumerName(), testStorage.getNamespaceName(), testStorage.getMessageCount());
+        // TODO: continuous clients are failing due to missing messages (one or two) during the whole process. This happens from time to time in upgrade/downgrade tests as well
```

Comment:
I am curious why they are failing here? :)

Reply:
I guess that's because in the moment the leader is rolled, the particular message is send - and the producer fails to send the message. IIRC I was trying to fix it for upgrade/downgrade as well - because it seems that there can be issue with configuration of the clients (that's what I remember from discussion with Tom Bentley, but it's quite a long time). So I'm disabling this check until I find the issue (and fix it).

Reply:
I don't know what "failing" here means but take into account that during brokers rolling you can expect transient errors, producer not able to send, consumer not able to receive (i.e. disconnection, leader partition changing, and more). But it has to recover every time as it happened with my manual tests. No messages are lost.

Reply:
Yeah that's what I actually mean, the producer in our test-clients just tries to send it, it catches some kind of exception and automatically decrease the number of successfully sent messages. So I need to look at this and fix that before enabling this check. Thanks Paolo